src/lt/objs/deploy.cljs

```diff
@@ -21,31 +21,64 @@
 
 (def shell (load/node-module "shelljs"))
 (def fs (js/require "fs"))
+; Update the comment for the following to describe the full situation:
+(def original-fs (js/require "original-fs")) ;; Required per [fs - reading asar files fails · Issue #1658 · atom/electron](https://github.com/atom/electron/issues/1658)
 (def zlib (js/require "zlib"))
 (def request (load/node-module "request"))
-(def tar (load/node-module "tar"))
+(def tar (load/node-module "tar-fs"))
+(def zip (load/node-module "node-stream-zip"))
 (def home-path (files/lt-home ""))
 ;; TODO: get-proxy
 ;; (def get-proxy (.-App.getProxyForURL (js/require "nw.gui")))
 (def get-proxy)
 
-(defn tar-path [v]
+; TODO: Cleanup platform name code
+
+; TODO: Rename to release-file-extension
+(def tar-extension (if (platform/win?) ".zip" ".tar.gz"))
+
+(defn get-release-download-url
+  "Retrieve the download URL for the specified release version and this computer's platform."
+  [v cb]
+  (let [release-info-url (str "https://api.github.com/repos/LightTable/LightTable/releases/tags/"
+                              v)]
+    (fetch/xhr release-info-url
+               {}
+               (fn [data]
```

Comment:
I'm not against your suggestion, and it's definitely much more idiomatic Clojure, but it's much harder to read for me. You also subtly changed the behavior; did you deliberately remove using the version (`v`)? Doing so is *probably* (?) okay, as I don't think we even have (or want to provide) a way to upgrade to any version that's not the latest, my version was (probably) intended to explicitly use the version specified.